src/htm/algorithms/Connections.cpp

```diff
@@ -90,17 +88,11 @@ Segment Connections::createSegment(const CellIdx cell,
   }
 
   //proceed to create a new segment
-  Segment segment;
-  if (!destroyedSegments_.empty() ) { //reuse old, destroyed segs
-    segment = destroyedSegments_.back();
```

Comment:
replacing the destroyed_ buffers. Makes code less complicated and the performance is the same.

Reply:
So this will never reuse a destroyed segment?

IMO: in a typical application, segments should never be destroyed, just created.  If the HTM reaches full capacity and needs to make room for new information, then the HTM should remove synapses from a segment instead of removing the segment itself.

Reply:
> So this will never reuse a destroyed segment?
IMO: in a typical application, segments should never be destroyed, just created.

yes. if you want, I can move this commit to a separete PR to discuss there? You're right we almost never remove segments/synapses (cca 0.X% on hotgym benchmark). 

> If the HTM reaches full capacity and needs to make room for new information, then the HTM should remove synapses from a segment instead of removing the segment itself.

if that is valid requirement, we should remove altogether the method `destroySegment()`. 

> HTM reaches full capacity and needs to make room

when is this? When synapses on a segment are full, when synapses on all segments are maxed, ...? 

> should remove synapses from a segment instead of removing the segment itself.

then we'd need to limit this pruning to some reasonable value (keep more synapses than activation threshold), otherwise the seg can never be active again. 
- currently we do this the other way, that such segment dies off (which I think is more plausible than setting a limit on a number of syns that need be left).